src/libsystemd-network/dhcp6-option.c

```diff
@@ -647,8 +647,10 @@ int dhcp6_option_parse_domainname(const uint8_t *optval, uint16_t optlen, char *
         _cleanup_strv_free_ char **names = NULL;
         int r;
 
-        assert_return(optlen > 1, -ENODATA);
-        assert_return(optval[optlen - 1] == '\0', -EINVAL);
+        if (optlen <= 1)
+                return -ENODATA;
+        if (optval[optlen - 1] != '\0')
+                return -EINVAL;
```

Comment:
`dhcp6_option_parse_domainname` can be reached through two paths: from dhcp6_lease_set_domains and dhcp6_lease_set_ntp which do somewhat different verification of parameters. But yeah, this is reasonable.

Reply:
Yes, a list of domains must always end with the zero label while a single domain doesn't have to (but can). I think this is the difference you are referring to, right?